influxvar.go

```diff
@@ -41,5 +53,39 @@ func NewStatistics(key, name string, tags map[string]string) *expvar.Map {
 	statMap.Init()
 	m.Set("values", statMap)
 
+	// Create a reference counter that we can use to release the map.
+	referencesVar := &expvar.Int{}
+	referencesVar.Set(2) // once for the site of usage, once for the monitor
+	m.Set("references", referencesVar)
 	return statMap
 }
+
+// Iterate over all the statistics maps.
+func DoStatistics(fn func(expvar.KeyValue)) {
+	root.Do(fn)
+}
+
+// Used to deregister a statistic when it is no longer needed.
+func DeleteStatistics(key string) {
```

Comment:
The consumers of DeleteStatistics seem to be separately tracking `statKey` in addition to `statMap`. I wonder if there's a clean way to change the API to pass `statMap` instead of `statKey` to DeleteStatistics.